### DISCUSSION TITLE
Updating phone number of user that has MFA enrolled

### DISCUSSION THREAD

**Discussion by urimext** [firebase/flutterfire#10102] [Q&A]
I'm trying to update phone number on user(**that has MFA enrolled**) through method: `FirebaseAuth.instance.currentUser?.updatePhoneNumber(credential)` but it fails with: 
_Unhandled Exception: [firebase_auth/unsupported-first-factor] Enrolling a second factor or signing in with a multi-factor account requires sign-in with a supported first factor. [ A phone number cannot be set as a first factor on an SMS based MFA user. ]_

It works in case when updating phone number of user that doesn't have MFA enrolled.

So is this restricted from Firebase? Or is there any other way to have both MFA enrolled and user with filled field "phoneNumber"?

Note: Through Admin SDK and Firebase console I can have both MFA & phoneNumber field filled.

**Comment by BlindDev** [comment]
@urimext have you found any solution yet?

**Comment by rafaelcolladojr** [comment]
While the Admin SDK does allow us to define MFA phone numbers, this method still requires a third-party OTP solution.

The frontend SDKs (eg. Flutter) should allow for an OTP-powered MFA phone number update solution.

**Comment by kristoflievens** [comment]
To update the phone number that is used as second factor, proceed as follows:
1) login the user again with MFA (e.g. use signInWithEmailAndPassword for email as first factor, followed by verifyPhoneNumber for the second factor)
2) when the MFA login successfully completed, enroll the new phone number (again using verifyPhoneNumber, but with the new phonenumber as input parameter)
3) when the new phone number is successfully added, unenroll the original phone number (using the MultiFactor.unenroll -> https://pub.dev/documentation/firebase_auth/latest/firebase_auth/MultiFactor/unenroll.html)

More details on step 1) and 2) can be found at https://firebase.google.com/docs/auth/flutter/multi-factor
